sqlglot/dialects/tsql.py

```diff
@@ -42,11 +51,24 @@ def _format_time(args):
     return _format_time
 
 
+def parse_format(args):
+    fmt = list_get(args, 1)
+    number_fmt = fmt.name in TRANSPILE_SAFE_NUMBER_FMT or not DATE_FMT_RE.search(fmt.this)
+    if number_fmt:
+        return exp.NumberToStr(this=list_get(args, 0), format=fmt)
+    return exp.TimeToStr(this=list_get(args, 0), format=exp.Literal.string(format_time(fmt.name, TSQL.time_mapping)))
+
+
 def generate_date_delta_with_unit_sql(self, e):
     func = "DATEADD" if isinstance(e, exp.DateAdd) else "DATEDIFF"
     return f"{func}({self.format_args(e.text('unit'), e.expression, e.this)})"
 
 
+def generate_format_sql(self, e):
+    fmt = e.args["format"] if isinstance(e, exp.NumberToStr) else exp.Literal.string(self.format_time(e.text("format")))
+    return f"FORMAT({self.format_args(e.this, fmt)})".replace("Y", "y")
```

Comment:
```suggestion
    return f"FORMAT({self.format_args(e.this, fmt)})".lower()
```

Reply:
mm != MM

Reply:
SELECT FORMAT (GETDATE(), 'dd MM yyy') =>28 10 2022
SELECT FORMAT (GETDATE(), 'dd mm yyy') => 28 03 2022 (minutes)

Reply:
this should just be handled in time_mapping then, time_mapping should return Y as y, maybe switch the ordering of time_mapping?

Reply:
same goes for the 1 letter formats, u, U , g, G etc.

Reply:
https://join.slack.com/t/tobiko-data/shared_invite/zt-1gw56z9k5-XnobJdCffsu0bWXV0PNwiA 

you can join my slack if you wanna chat, maybe easier

Reply:
Hi Toby,

Please review the solution I went with relying on order (feels wierd I always pushed people to not rely on it but python has evolved :)) and took the single letters as seperate time mapping.